pdns/credentials.cc

```diff
@@ -156,3 +160,89 @@ bool CredentialsHolder::isHashingAvailable()
   return false;
 #endif
 }
+
+#include <signal.h>
+#include <termios.h>
+
+std::string CredentialsHolder::readFromTerminal()
+{
+  struct termios term;
+  struct termios oterm;
+  memset(&term, 0, sizeof(term));
+  term.c_lflag |= ECHO;
+  memset(&oterm, 0, sizeof(oterm));
+  oterm.c_lflag |= ECHO;
```

Comment:
Won't this be overwritten by `tcgetattr`?

Reply:
but if it doesn't (because somebody breaks this code in the future), this will cause restoreTermSettings, if it happens when it shouldn't, to at least restore ECHO.

Reply:
I kinda like the structrure in the original where term and oterm are only initialised if input != stdin or tcgetattr fails better

Reply:
OK, I'll see what I can do, hopefully next week.